### Welcome to the Final Project!

##### import Pandas and other libraries

In [ ]:
import pandas as pd 
import numpy as np 
import psycopg2

#### Here is our Dataset

In [ ]:
weather_df = pd.read_csv('http://rattle.togaware.com/weatherAUS.csv')

In [ ]:
weather_df.info()

In [ ]:
print(list(weather_df.columns))

In [ ]:
print(max(weather_df.Date))
print(min(weather_df.Date))

In [35]:
weather_df.apply(lambda x: x['MinTemp'] + x['MaxTemp'], axis=1)
weather_df.Location.map(lambda x: x)

Date Location  MinTemp  MaxTemp  Rainfall  Evaporation  Sunshine  \
0  2008-12-01   Albury     13.4     22.9       0.6          NaN       NaN   
1  2008-12-02   Albury      7.4     25.1       0.0          NaN       NaN   
2  2008-12-03   Albury     12.9     25.7       0.0          NaN       NaN   
3  2008-12-04   Albury      9.2     28.0       0.0          NaN       NaN   
4  2008-12-05   Albury     17.5     32.3       1.0          NaN       NaN   

  WindGustDir  WindGustSpeed WindDir9am  ... Humidity3pm  Pressure9am  \
0           W           44.0          W  ...        22.0       1007.7   
1         WNW           44.0        NNW  ...        25.0       1010.6   
2         WSW           46.0          W  ...        30.0       1007.6   
3          NE           24.0         SE  ...        16.0       1017.6   
4           W           41.0        ENE  ...        33.0       1010.8   

   Pressure3pm  Cloud9am  Cloud3pm  Temp9am  Temp3pm  RainToday  RISK_MM  \
0       1007.1       8.0       NaN     16.9     21.8         No      0.0   
1       1007.8       NaN       NaN     17.2     24.3         No      0.0   
2       1008.7       NaN       2.0     21.0     23.2         No      0.0   
3       1012.8       NaN       NaN     18.1     26.5         No      1.0   
4       1006.0       7.0       8.0     17.8     29.7         No      0.2   

   RainTomorrow  
0            No  
1            No  
2            No  
3            No  
4            No  

[5 rows x 24 columns]

#### Here is what we are trying to achieve 

1. First, make your initial data exploration before starting the following task.
1. Select only `Date`,`location`, `MinTemp`, `MaxTemp`, `Rainfall`, `Sunshine`, `RainToday` fields that are after `2010`.
1. Rename all the columns to something easier to read and understand.
1. Then change the `RainToday` values (of cases after you renamed the column) to `0` when is `No` and `1` when it is `Yes`.
1. Add a new column `day`, `month`, and `year` column (three new columns) based on `Date` column.
1. Fill all missing `MinTemp` and `MaxTemp` values as follows:
  1. calculate the `yearlly_average` minimum and maximum temperature
  2. then fill the missing values by what you calculate as the yearly average minimum and maximum temperature.

      example: 
        - say the average `2020` temperature was `16^c`. so this means any 2020 record that missed a value would have been replaced by the yearly average minimum temperature value of `16^c`.
        - the same goes for the maximum temperature column. 
1. Finally, add the degree centigrade symbol for both the minimum and the maximum temperates columns.
  1. i.e 22.9 will be `22.9^c`
1. Lastly, create a table in your Postgresql and write this data to your database table. 

In [33]:
def good_luck(item):
    if item == 'question':
        output = "Fell free to {} me or {} me in slak or schedule a zoom meeting to go over staffs.".format('IM', 'CALL')
    elif 'comment' in item or 'suggession' in item:
        output = "Fell free to directely send me your {} or {}.".format(item.split('/')[0], item.split('/')[1])
    else:
        output = "I wish you all the best and Good luck on  your journey."
    
    return output

In [34]:
feedback_list = ['question', 'comment/suggession', 'otherwise']
for feedbak in feedback_list:
    print(good_luck(feedbak))

Fell free to IM me or CALL me in slak or schedule a zoom meeting to go over staffs.
Fell free to directely send me your comment or suggession.
I wish you all the best and Good luck on  your journey.


In [ ]:
import random

def select_random_Ns(lst, n):
    random.shuffle(lst)
    result = []
    for i in range(0, len(lst), n):
        result.append(lst[i:i + n])
    return result
        
        
lst = [1, 2, 3, 4, 5, 6, 7, 8, 9]

print(select_random_Ns(lst, 3))

In [ ]:
et_timezone = ['Helina','Kesete','Mihret','Salm','Athnafu', 'Hiwot', 'Milkiyas','Bantamlak','Seble','Muluneh','Fantahun']
print(len(et_timezone))

In [ ]:
select_random_Ns(et_timezone, 3)

# [['Milkiyas', 'Helina', 'Fantahun'],
#  ['Seble', 'Bantamlak', 'Athnafu'],
# 1.0  ['Kesete', 'Hiwot', 'Mihret'],
#  ['Muluneh', 'Salm']]

In [ ]:
us_timezone = ['Ehite','Abinet','Baheru','Betty','Beza','Biniyam','Biz','Fisum','Hana','Hayat','Hilliwuna',
    'Abdu','Jerusalem','Mululbrhan', 'Semhal', 'Wendy','Zelalem','Kidist','Afnan','Samrawit','Tizazu','Fithum',
    'Dagew','Gelila','Tilahun','Hana Abera']
print(len(us_timezone))

In [ ]:
select_random_Ns(us_timezone, 3)
# [3. ['Biniyam', 'Biz', 'Wendy'],
#  2. ['Abdu', 'Zelalem', 'Jerusalem'],
#  ['Samrawit', 'Ehite', 'Mululbrhan'],
#  5. ['Semhal', 'Afnan', 'Tilahun'],
#  4. ['Beza', 'Hilliwuna', 'Hayat'],
#  ['Hana Abera', 'Fithum', 'Hana'],
#  ['Dagew', 'Abinet', 'Tizazu'],
#  ['Kidist', 'Fisum', 'Gelila'],
#  ['Baheru', 'Betty']]